webring/webring.py

```diff
@@ -78,35 +80,34 @@ def get_pelican_settings(generators):
     return generators[0].settings
 
 
-def setup_feedparser(settings):
-    feedparser.USER_AGENT = 'Webring Pelican plugin/%s ' \
-        '+https://github.com/pelican/pelican-plugins' % WEBRING_VERSION
-
-    if settings[WEBRING_SUPPORT_HTTPS_STR]:
-        # https://www.python.org/dev/peps/pep-0476/#opting-out
-        import ssl
-        try:
-            _create_unverified_https_context = ssl._create_unverified_context
-        except AttributeError:
-            # Legacy Python that doesn't verify HTTPS certificates by default
-            pass
-        else:
-            # Handle target environment that doesn't support HTTPS verification
-            ssl._create_default_https_context = \
-                _create_unverified_https_context
-
-
-def get_feed_articles(feed_url, settings):
-    parsed_feed = feedparser.parse(feed_url)
+def get_feed_html(feed_url):
+    try:
+        req = Request(feed_url)
+        req.add_header(
+            'User-Agent',
+            (
+                'Webring Pelican plugin/{} '
+                + '+https://github.com/pelican/pelican-plugins'
+            ).format(WEBRING_VERSION)
+        )
+        return urlopen(req).read().decode('utf-8')
```

Comment:
What would happen in case of a non-200 HTTP status code ?

Reply:
Error status codes returned by the server (3xx and 4xx) are handled in the following lines by catching `URLError`. Is that what you refered to?

Reply:
Yes. I wasn't sure if an exception would be raised. Thanks for your answer

Reply:
Thank you for your review @Lucas-C :tada: